src/libexpr/def-expr.cc

```diff
@@ -0,0 +1,13 @@
+#include "def-expr.hh"
```

Comment:
I think the cases in which we fish files out of the user's file system are of high importance and deserve to be called out explicitly in the code.

To the extent that def-expr has structure, we can put more things here. The documentation would indicate it has some some structure, so we can do that.

Also, it ideally wouldn't be moved out of `libexpr` as this is more a matter of how the language is *used* than what it *is*. Giving it a separate home like this makes it easy to migrate it.

Reply:
In essence, consider this: we all agree defexpr is a bad all pre new heremetic stuff thing --- I am giving it a new module because I don't want it to "pollute" anything else!